src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -74,6 +76,7 @@ internal void RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment re
         }
 
         _nextRenderTree.Clear();
+        _nextRenderTree.TrackNamedEventHandlers = _renderer.ShouldTrackNamedEventHandlers();
```

Comment:
We don't do work if you call `SetEventHandlerName` but the renderer is not tracking the event handler names.

Reply:
Actually, I did this for another reason. Tracking event handler names (since we added them to forms) might result in us throwing an exception if two forms end up with the same name, so I want to eliminate any chance that this breaks any existing app that is not ready for handling forms server-side, and I want to eliminate the chance that having duplicate event names precludes the app from rendering, hence why I choose to limit the tracking to when we are trying to dispatch an event (which is the only time when we **really** need to make a decision).